dashboard/src/views/Projects/index.tsx

```diff
@@ -0,0 +1,120 @@
+import SearchIcon from "@mui/icons-material/Search";
+import { Box, FormControl, FormLabel, Input, Sheet, Table } from "@mui/joy";
+import { useEffect, useState } from "react";
+import { mockedProjectsResponse } from "../../api/mocks/projects.mock";
+import { getProjectsResponse } from "../../interfaces/projects";
+import { PaginationSection } from "./components/PaginationSection/PaginationSection";
+import { ProjectTableRow } from "./components/Row";
+
+export const ProjectsOverview = () => {
+  const [searchBar, setSearchBar] = useState("");
+  //TODO: replace mocked data with the response from the API when react query hooks are implemented
+  const projects = mockedProjectsResponse;
+  const [projectsList, setProjectsList] = useState<getProjectsResponse>(
+    mockedProjectsResponse,
+  );
+
+  useEffect(() => {
+    if (searchBar === "") {
+      return;
+    }
+    setProjectsList(projectsList);
+  }, [projectsList, searchBar]);
```

Comment:
this could cause an infinite loop

Reply:
I was concerned about that as well but getting a warning that I'm missing a dependency if I'm deleted either of them

Reply:
i dotn think you should delete them, we just need to look at the prev and only update if there is a change. We can look at this tomorrow.